File: 3636-check-balanced-string/3636-check-balanced-string.cs
public class Solution {
    public bool IsBalanced(string num) {
        var sum = 0;
        for (var i = 0; i < num.Length; ++i) {
            var dig = num[i] - '0';
            if ((i & 1) != 0)
                sum += dig;
            else
                sum -= dig;
        }
        return sum == 0;
    }
}

File: 3636-check-balanced-string/3636-check-balanced-string.java
class Solution {
    public boolean isBalanced(String num) {
        int sum = 0;
        for (int i = 0; i < num.length(); ++i)  {
            if ((i & 1) != 0) 
                sum += (num.charAt(i) - '0');
            else 
                sum -= (num.charAt(i) - '0');
        }
        return sum == 0;
    }
}